tests/test_dummy.py

```diff
@@ -0,0 +1,309 @@
+    assert_anchor_group_equality(groups_of_1, [[1]])
+    assert_anchor_group_equality(groups_of_2, [[1, 0], [1, 2]])
+    assert_anchor_group_equality(groups_of_3, [[1, 0, 2], [1, 2, 0]])
+
+    groups_of_1, groups_of_2, groups_of_3 = identify_anchor_groups(mol, core, 2)
+    assert_anchor_group_equality(groups_of_1, [[2]])
+    assert_anchor_group_equality(groups_of_2, [[2, 1], [2, 0]])
+    assert_anchor_group_equality(groups_of_3, [[2, 1, 0], [2, 0, 1]])
+
+    mol = Chem.MolFromSmiles("NCCF")
+    core = [1, 2]
+    groups_of_1, groups_of_2, groups_of_3 = identify_anchor_groups(mol, core, 1)
+    assert_anchor_group_equality(groups_of_1, [[1]])
+    assert_anchor_group_equality(groups_of_2, [[1, 2]])
+    assert_anchor_group_equality(groups_of_3, [])
+
+    mol = Chem.MolFromSmiles("C(C)(C)(C)C")
+    core = [0]
+    groups_of_1, groups_of_2, groups_of_3 = identify_anchor_groups(mol, core, 0)
+    assert_anchor_group_equality(groups_of_1, [[0]])
+    assert_anchor_group_equality(groups_of_2, [])
+    assert_anchor_group_equality(groups_of_3, [])
+
+    mol = Chem.MolFromSmiles("C(C)(F)CC")
+    core = [1, 0, 4, 3]
+    groups_of_1, groups_of_2, groups_of_3 = identify_anchor_groups(mol, core, 0)
+    assert_anchor_group_equality(groups_of_1, [[0]])
+    assert_anchor_group_equality(groups_of_2, [[0, 1], [0, 3]])
+    assert_anchor_group_equality(groups_of_3, [[0, 3, 4]])
+
+
+def test_enumerate_anchor_groups():
+    mol = Chem.MolFromSmiles("C1CC2OOC12")
+    core = [0, 1, 2, 5]
+    groups_of_1, groups_of_2, groups_of_3 = enumerate_anchor_groups(mol, core, [3, 4])
+    assert_anchor_group_equality(groups_of_1, [[2], [5]])
+    assert_anchor_group_equality(groups_of_2, [[2, 1], [2, 5], [5, 0], [5, 2]])
+    assert_anchor_group_equality(groups_of_3, [[2, 1, 0], [2, 5, 0], [5, 0, 1], [5, 2, 1]])
+
+    # aspirin
+    mol = Chem.MolFromSmiles("CC(=O)OC1=CC=CC=C1C(=O)O")
+    core = [3, 4, 5, 6, 7, 8, 9, 10]
+    groups_of_1, groups_of_2, groups_of_3 = enumerate_anchor_groups(mol, core, [0, 1, 2, 11, 12])
+    assert_anchor_group_equality(groups_of_1, [[3], [10]])
+    groups_of_1, groups_of_2, groups_of_3 = enumerate_anchor_groups(mol, core, [0, 1, 2])
+    assert_anchor_group_equality(groups_of_1, [[3]])
+    assert_anchor_group_equality(groups_of_2, [[3, 4]])
+    assert_anchor_group_equality(groups_of_3, [[3, 4, 5], [3, 4, 9]])
+    groups_of_1, groups_of_2, groups_of_3 = enumerate_anchor_groups(mol, core, [11, 12])
+    assert_anchor_group_equality(groups_of_1, [[10]])
+    assert_anchor_group_equality(groups_of_2, [[10, 9]])
+    assert_anchor_group_equality(groups_of_3, [[10, 9, 8], [10, 9, 4]])
+
+
+def test_enumerate_allowed_dummy_ixns_3_anchors():
+    dummy_group = {0, 1, 5, 8}
+    anchor_group = [2, 3, 4]
+
+    allowed_ixns = enumerate_dummy_ixns(dummy_group, anchor_group)
+
+    for ixn in allowed_ixns:
+        assert len(ixn) == len(set(ixn))
+
+    assert (0, 1, 2, 3) in allowed_ixns
```

Comment:
Is this the complete set of ixns and if so would be good to assert of the number of allowed ixns?

Reply:
This is not the complete set of ixns, just a spot check. Enumerating the full set of `allowed_ixns` requires more skills in combinatorics than I currently possess.